### PR TITLE
Mechbay ruin

### PR DESCRIPTION
**Pull request by Erikafox**
## About The Pull Request

Adds a new ruin, the Abandoned Mech Bay. The abandoned mech bay is a semi-spacious installation formerly made to hold 4 mechs, and their crew. The base was designed to resist attack, but in its age is much more approachable. 

The air processing cave has a small amount of landmines and turrets, along with the quickest way in. Anyone with a weapon and basic medical supplies could destroy the turrets and enter through air processing. 
It is also possible to breach through where the walls are thin, such as at the south and west sides of the base.
Once inside you will have to deal with an assortment of syndicate simplemobs, and another set of turrets. If you can clear the main hallway and restore power, it's likely you can get inside the mech bays.
The base has fully mapped atmospherics and power supplies. If you really wanted to go so far, you could clean and live in it. 
Loot includes:
2 autolathes
2 Exosuit Fabricators
1 Intact Gygax
1 heavy mech laser
1 mech lmg
2 mech lmg ammo boxes
3 oxygen cans
3 nitrogen cans
Assorted Materials
Anything you can pick up off corpses
A medical aid kit
2 first aid kits
1 toxin aid kit
McKinnon the fox
2 Exosuit Diamond Drills
T2 mech parts
Industrial Welders
6 welding gas masks
Toolbelts
3 Pilot helmets
3 mech pilot suits (the funny NGE ones not spacesuits)
An R&D console
Anything you can get by disassembling the base.

- [ ] I affirm that I have tested all of my proposed changes and that any issues found during tested have been addressed.

## Why It's Good For The Game
More ruin variety is good. ~~It'll fill out the rock planets a bit more, and give something extra to waste planets when overmap 4 is merged.~~ Overmap 4 is now merged. This map is for waste worlds and has been updated for them. It should present an interesting challenge while being mildly thematic to the planets.
Mechs are also very cool.

## Changelog

:cl:
add: The Abandoned Mech Bay, a new ruin will start spawning on rock worlds. Beware its defenses, and those who would wish to salvage it before you.
add: Area defs for the Mech Bay
/:cl:

### CODE REVIEW

**Comment by Erikafox** [comment]
I need to do a go-over of the loot, decals, and possibly theme it towards a faction (Nanotrasen was suggested to me) before this is ready to be undrafted. Some feedback on what's there would be appreciated.

File: code/game/area/areas/ruins/rockplanet.dm [hunk 1]
```diff
@@ -6,3 +6,61 @@
 /area/mine/rockplanet_nanotrasen
 	name = "Abandoned Mining Facility"
 
+//Abandoned Mechbay Ruin
```
Comment (Latentish) [inline code/game/area/areas/ruins/rockplanet.dm]: This is way too many unique areas. Lower it down to like 1 or 2.

**Review by Latentish — CHANGES_REQUESTED** [review]
image
This is properly area'd yes?

I'm not sure about having an intact gygax. I'd be much more ok if you made it the Circuits and some of the pieces to build it. Could put a mech fab circuit somewhere, like instead of the rnd console.

File: code/game/area/areas/ruins/rockplanet.dm (hunk 1 above)
Comment (Erikafox) [reply code/game/area/areas/ruins/rockplanet.dm]: I can cut one or two out. It's fine otherwise.

**Comment by Erikafox** [comment]
>This is properly area'd yes?

Yes. See the updated screenshots.
 
> I'm not sure about having an intact gygax. I'd be much more ok if you made it the Circuits and some of the pieces to build it. Could put a mech fab circuit somewhere, like instead of the rnd console.

I don't think investigating some mysterious built into a mountain and seeing the parts to build something has the same impact as "HOLY SHIT A MECH", especially since the mech is supposed to be the ruin's "cool thing"

**Comment by Erikafox** [comment]
I've redone the map and areas to set this ruin up for Waste Worlds now that overmap 4 is merged. This also included adding a wasteworld ruin area file and fiddling around in the dme.

**Comment by ForestCranked** [comment]
god will not have mercy

**Comment by Latentish** [comment]
This needs to actually be tested on live. I literally had this tm'd and passed out for the whole day. I'm stupid and sleepy sometimes. I'll get this actually tested next uptime.

**Review by Latentish — APPROVED** [review]

**Comment by SweetBlueSylveon** [comment]
https://twitter.com/SolGgx/status/1588785707145134080?s=20&t=4hzh3DjmUv8hNzE7ZcrCYw

**Comment by Latentish** [comment]
image

**Comment by ZephyrTFA** [comment]
This was approved and then never merged, sounds like a skill issue to me

**Comment by Latentish** [comment]
> This was approved and then never merged, sounds like a skill issue to me

This was never testmerged.

**Comment by ZephyrTFA** [comment]
Approving a PR marks it as ready to merge

**Comment by Latentish** [comment]
> Approving a PR marks it as ready to merge

The comment directly above the approval said "This needs to get testmerged next uptime" then it had multiple changes afterwards, meaning that approval is somewhat null.

**Comment by ZephyrTFA** [comment]
If you need something to be test merged before full merge use the label

**Comment by Latentish** [comment]
> If you need something to be test merged before full merge use the label

image

**Comment by ZephyrTFA** [comment]
Wrong label

**Comment by Erikafox** [comment]
This is the only ruin I haven't seen get tested during uptime too I should add. It's been spawned in but not actually ran.

**Pull request merged**